_posts/2024-12-25-elm-after-7-years.md

```diff
@@ -0,0 +1,35 @@
+---
+layout: post
+title:  "What I learned from 7 years of Elm Development"
+date:   2024-12-25 12:00:00 -0500
+categories: blog
+author: Aaron
+author_twitter: shamshirz
+image: /assets/elm/elm.webp
+---
+My development perspective on using Elm, and when it will serve you. [![Elm-lang logo]({{ "/assets/elm/elm.webp" | relative_url }}){:class="excerpt-image"}](https://elm-lang.org)
+
+**Note about Shamshirz**
+
+*This post reflects my experience working extensively with [Elm](https://elm-lang.org) at a [B2B insurance startup](http://corvusinsurance.com/) from 2017 to acquisition in 2024, as the 2nd Engineer.*
+
+I'm looking at a new year of big changes and I want to share my appreciation and advocacy through reflection. Elm is my Gold Standard, best in class UI development kit. I had experience with SSR templating languages, vanilla JS sites, and React just prior to picking up Elm. Elm was my introduction to a pleasant type systems (👎 Java). The type system is friendly, clarifies thinking, and it provides the best feedback loop I’ve experienced in engineering. I learned that Elm can be rocket fuel and I hope to articulate the the conditions for success that we encountered. Elm isn’t without its limitations, but allow me to convince you why it's worth it, dear reader.
+
+## Why Elm Shines
+The type system is simple and delightful. The compiler is fast, capable, and descriptive enough to be a coding assistant without the need of an LLM. Problem solving is mostly modeling types to reflect your needs and the code writes itself. The biggest hurdle is adapting to the syntax the flow of data. The language is small! Once you are past basics, you can participate in any discussion and I always found it fun to iterate, learn, and improve. We even found that it was unnecessary to write tests. 
```

Comment:
> The language is small! Once you are past basics, you can participate in any discussion and I always found it fun to iterate, learn, and improve.

I feel like this might be a central theme to the whole post, and I'd love to see another sentence or two just on the implications of this. Like a comparison with Javascript where you have to at least be aware of the esoteric historical BS syntax or the class system or whatever to be able to understand any random person's code, but with Elm the code will look so similar to code you've seen because the language's surface area is SMOL and its usage is pretty uniform (there's even a standardized style guide / formatter). And below when you say the language is "boring" technology, I feel like that's another place you can refer to the smolness - it's boring b/c it's well-known and it CAN be well-known even with the comparatively lesser battle-testing b/c it's SMOL - there's fewer things to battle test.